### PR TITLE
Improve unit test coverage of scale

### PR DESCRIPTION
**Pull request by yuexiao-wang**
**What this PR does / why we need it**:
1. Add unit test of ReplicaSet and StatefulSet to improve coverage of scale
2. Fix variable "fake" collideds with imported package name
3. Use Error with no value format and fix typo error messages

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubectl/scale_test.go [hunk 1]
```diff
@@ -785,3 +787,527 @@ func TestValidateDeployment(t *testing.T) {
...
+	scaler.Scale("default", name, count, &preconditions, nil, nil)
+
+	actions := faker.Actions()
+	if len(actions) != 2 {
+		t.Errorf("unexpected actions: %v, expected 2 actions (get, update)", actions)
+	}
+
+	if action, ok := actions[0].(testcore.GetAction); !ok || action.GetResource().GroupResource() != apps.Resource("statefulsets") || action.GetName() != name {
+		t.Errorf("unexpected action: %v, expected get-statefulsets %s", actions[0], name)
+
```
Comment (brendandburns) [inline pkg/kubectl/scale_test.go]: delete empty line.

File: pkg/kubectl/scale_test.go [hunk 2]
```diff
@@ -63,8 +65,8 @@ func (c *ErrorReplicationControllerClient) ReplicationControllers(namespace stri
 }
 
 func TestReplicationControllerScaleRetry(t *testing.T) {
-	fake := &ErrorReplicationControllerClient{Clientset: fake.NewSimpleClientset(oldRc(0, 0)), conflict: true}
-	scaler := ReplicationControllerScaler{fake}
+	faker := &ErrorReplicationControllerClient{Clientset: fake.NewSimpleClientset(oldRc(0, 0)), conflict: true}
```
Comment (brendandburns) [inline pkg/kubectl/scale_test.go]: no need to change the name here, please revert to `fake` so that we can minimize difs.

**Comment by brendandburns** [comment]
/retest

This basically LGTM, a couple of small nits.

Thanks for the added test coverage!

File: pkg/kubectl/scale_test.go (hunk 1 above)
Comment (yuexiao-wang) [reply pkg/kubectl/scale_test.go]: Got it

**Comment by yuexiao-wang** [comment]
@brendandburns Thanks for reviews and the PR is updated

**Comment by yuexiao-wang** [comment]
ping @brendandburns

**Comment by yuexiao-wang** [comment]
@brendandburns The PR has updated

**Comment by yuexiao-wang** [comment]
@brendandburns Any suggestion about this PR?

**Comment by yuexiao-wang** [comment]
ping @brendandburns

**Pull request merged**